docs/en/stack/ml/nlp/ml-nlp-categorize-text.asciidoc

```diff
@@ -0,0 +1,160 @@
+classification, where the number of classes is exactly two, and
+multi-class classification, where the number of classes is more than
+two.
+
+This task, for example, can help you analyze text for markers of positive or 
+negative sentiment or classify text into various topics.
+
+The example below is of binary classification where the model is trained to 
+perform sentiment analysis with classes "POSITIVE" and "NEGATIVE".
+
+[source,js]
+----------------------------------
+...
+{
+    "input_text": "This was the best movie I’ve seen in the last decade!"
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+The task returns the following result:
+
+[source,js]
+----------------------------------
+...
+{
+    "result": "POSITIVE"
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+The second example is of multi-class classification where the model is trained 
+to classify text into news topics with classes "SPORTS", "BUSINESS", "LOCAL", 
+and "ENTERTAINMENT".
+
+[source,js]
+----------------------------------
+...
+{
+    "input_text": "The Blue Jays played their final game in Toronto last night and came out with a win over the Yankees, highlighting just how far the team has come this season."
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+The task returns the following result:
+
+[source,js]
+----------------------------------
+...
+{
+    "result": "SPORTS"
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+[discrete]
+[[ml-nlp-zero-shot]]
+== Zero-shot text classification
+
+The zero-shot classification task offers the ability to classify text without 
+training a model on a specific set of classes. It uses a model trained on a 
+large dataset that has gained a general language understanding and asks the 
+model how well the labels you provided fit with your text.
+
+This task enables you to analyze and classify your input text even when you 
+don't have sufficient training data to train a text classification model.
+
+Let's see an example:
+
+[source,js]
+----------------------------------
+...
+{
+    "input_text": "The S&P 500 gained a meager 12 points in the day’s trading. Trade volumes remain consistent with those of the past week while investors await word from the Fed about possible rate increases.",
+    "labels": ["SPORTS", "BUSINESS", "LOCAL", "ENTERTAINMENT"]
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+The task returns the following result:
+
+[source,js]
+----------------------------------
+...
+{
+    "result": "BUSINESS"
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+Note that – unlike in the case of the text classification task above – the model 
+that is used here is not trained specifically for news categorization; in fact, 
+the possible labels are provided together with the input text at inference time. 
+This task allows you to use the same model to perform inference on a different 
```

Comment:
If you agree with my earlier suggestion, this could be reduced to something like this (I don't think it matters whether it's the same or different input text, so I'd recommend omitting that bit) and omit the pseudo-results here too:

> You can use the same model to perform inference with different classes, such as: